### PR TITLE
Documentation cleanup, fixed new_term example, include all

### PR DESCRIPTION
**Pull request by Qmando**
1. Changes the table in the documentation to be single width. Also added some missing options.
2. Changed include to default to "*". This is the number one thing that confuses people. Including
all the fields is an expected behavior.
3. Change new_term example rule. The example was broken #172 
4. Added note about time format
5. Changed all rule options to be subsections. This still requires some cleanup because all I did was add headers.
6. Clarified what a filter does
7. Clarified new_term and query_key
8. Fixed some optional settings being under the required section

Tested rendered page locally, looks alright. Not the same as readthedocs rendered.

### CODE REVIEW

**Comment by talyian** [comment]
lg2m

File: docs/source/ruletypes.rst [hunk 1]
```diff
@@ -99,56 +139,98 @@ Common Configuration Options
...
+``alert``: The ``Alerter`` type to use. This may be one or more of the built in alerts, see :ref:`Alert Types <alerts>` section below for more information,
+or loaded from a module. For loading from a module, the alert should be specified as ``module.file.AlertName``. (Required, string or list, no default)
 
 Optional Settings
 ~~~~~~~~~~~~~~~~~
 
+use_ssl
+^^^^^^^
+
+``use_ssl``: Whether or not to connect to ``es_host`` using SSL. (Optional, boolean, default F)
```
Comment (y-trobinso) [inline docs/source/ruletypes.rst]: If this defaults "False", just call it that. "F" is misleading.

File: docs/source/ruletypes.rst [hunk 2]
```diff
@@ -163,23 +248,42 @@ at 1:15, the next alert will not be until at least 1:35. If another alert fires
...
 All of the results of querying with these filters are passed to the ``RuleType`` for analysis.
 For more information writing filters, see :ref:`Writing Filters <writingfilters>`. (Required, Elasticsearch query DSL, no default)
 
-``include``: A list of terms that should be included in query results and passed to rule types and alerts. '@timestamp', ``query_key``,
-``compare_key``, and ``top_count_keys``  are automatically included, if present. (Optional, list of strings)
+include
+^^^^^^^
+
+``include``: A list of terms that should be included in query results and passed to rule types and alerts. When set, only those
+fields, along with '@timestamp', ``query_key``, ``compare_key``, and ``top_count_keys``  are included, if present.
```
Comment (y-trobinso) [inline docs/source/ruletypes.rst]: Too much whitespace after ``top_count_keys``

File: example_rules/example_new_term.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# Alert when a login event is detected for a never before seen combination of (username, computer)
-# In this example, "login" logs contain which user has logged into what computer
+# Alert when a login event is detected for user "admin" never before seen ip
+# In this example, "login" logs contain which user has logged in from what IP
```
Comment (y-trobinso) [inline example_rules/example_new_term.yaml]: This prose uses "ip" and "IP" interchangeably. Please use just one capitalization.

File: docs/source/ruletypes.rst [hunk 4]
```diff
@@ -5,41 +5,81 @@ Examples of several types of rule configuration can be found in the example_rule
...
++----------------------------------------------------+           |
+| ``index`` (string)                                 |           |
++----------------------------------------------------+           |
+| ``name`` (string)                                  |           |
++----------------------------------------------------+           |
+| ``type`` (string)                                  |           |
++----------------------------------------------------+           |
+| ``alert`` (string or list)                         |           |
++----------------------------------------------------+-----------+
+| ``use_strftime_index`` (boolean, default F)        |  Optional |
```
Comment (y-trobinso) [inline docs/source/ruletypes.rst]: Please write out "False" or "True". Calling these "T" and "F" is misleading.

**Comment by y-trobinso** [comment]
lgtm, once the above pedantic style concerns are fixed

**Pull request merged**